lib/connection.js

```diff
@@ -639,11 +639,12 @@ Connection.prototype.openUri = function(uri, options, callback) {
       this.pass = options.auth.password;
     }
 
+    // Mongoose has its own internal buffering mechanism by default
     if (options.bufferCommands != null) {
-      throw new Error('options.bufferCommands not supported on this custom fork of mongoose for Gitter');
-      // options.bufferMaxEntries = 0;
-      // this.config.bufferCommands = options.bufferCommands;
-      // delete options.bufferCommands;
+      // Since Mongoose has its own buffering system, disable the MongoDB buffering
```

Comment:
Explained at https://mongoosejs.com/docs/4.x/docs/guide.html#bufferCommands,

> By default, mongoose buffers commands when the connection goes down until the driver manages to reconnect.

and https://mongoosejs.com/docs/4.x/docs/connections.html#options

> `bufferMaxEntries` - The MongoDB driver also has its own buffering mechanism that kicks in when the driver is disconnected. Set this option to 0 and set bufferCommands to false on your schemas if you want your database operations to fail immediately when the driver is not connected, as opposed to waiting for reconnection.

---

Also see https://github.com/mongodb/node-mongodb-native/blob/ef895c4dd26c6f8069dc2cc4c948f404b4daad0c/docs/content/tutorials/connection_failures.md for a  good overview of the buffer stuff in the MongoDB Driver